src/eth/TaraClient.sol

```diff
@@ -184,62 +183,17 @@ contract TaraClient is IBridgeLightClient {
         }
     }
 
-    function setPending(PillarBlock.WithChanges memory b, bytes32 ph) internal {
-        pending.blockWithChanges = b;
-        pending.blockHash = ph;
-        pending.finalized = false;
-    }
-
-    /**
-     * @dev Adds a pending block
-     * @param b RLP encoded PillarBlockWithChanges struct representing the pending block.
-     */
-    function addPendingBlock(bytes memory b) public {
-        PillarBlock.WithChanges memory pb = PillarBlock.fromBytes(b);
-        bytes32 ph = PillarBlock.getHash(b);
-
-        require(block.number >= finalized.finalizedAt + delay, "The delay has not passed yet");
-
-        if (pending.finalized) {
-            require(pb.block.prevHash == finalized.blockHash, "Block prevHash != finalized block hash");
-            setPending(pb, ph);
-            return;
-        }
-
-        require(pb.block.prevHash == pending.blockHash, "Block prevHash != pending block hash");
-        finalized = PillarBlock.FinalizedBlock(pending.blockHash, pending.blockWithChanges.block, block.number);
-        processValidatorChanges(pending.blockWithChanges.validatorChanges);
-        setPending(pb, ph);
-    }
-
-    /**
-     * @dev Finalizes a pending block by providing the required signatures.
-     * @param signatures The array of signatures required to finalize the block.
-     */
-    function finalizePendingBlock(CompactSignature[] memory signatures) public {
-        _finalizeBlock(pending.blockWithChanges, pending.blockHash, signatures);
-        pending.finalized = true;
-    }
-
     /**
      * @dev Finalizes
      * @param b The PillarBlockWithChanges struct containing the block data and validators changes.
-     * @param signatures An array of signatures.
+     * @param h Pillar block hash
      */
-    function _finalizeBlock(PillarBlock.WithChanges memory b, bytes32 h, CompactSignature[] memory signatures)
-        internal
-    {
-        uint256 gasleftbefore = gasleft();
+    function _finalizeBlock(PillarBlock.WithChanges memory b, bytes32 h) internal {
```

Comment:
do we need this whole function ? 

- This check is already in finlizeBlocks function: `require(b.block.prevHash == finalized.blockHash, "block.prevHash != finalized.blockHash");`
- This one should be right after it so we don't even process vote counts changes etc... ```
require(
            b.block.period == finalized.block.period + pillarBlockInterval,
            "Pending block should have number pillarBlockInterval greater than latest"
        );
```
and then the last line can be put into the finalizeBlocks function too: `finalized = PillarBlock.FinalizedBlock(h, b.block, block.number);`

Reply:
This first check got accidentally there. So I will prefer to have a finalize and remove this first check, because ` so we don't even process vote counts changes` doesn't make a huge change I believe. @VargaElod23?

Reply:
Well if we can, then let's save gas by moving it ahead. @kstdl